src/kernels/MIOpenCat.cpp

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+
+//#if MIOPEN_USE_BFP16 == 1
+//#undef FLOAT
+//#define FLOAT hip_bfloat16
+//#endif
+
+template <typename T>
+__device__ void cat_copy_buf(const T* __restrict__ input,
+                             T* __restrict__ output,
+                             const size_t input_dim_size,
+                             const int stride,
+                             size_t* output_offset)
+{
+    if(!input)
+        return;
+
+    int gid0            = blockIdx.x * blockDim.x + threadIdx.x;
+    int gid1            = blockIdx.y * blockDim.y + threadIdx.y;
+    int gsz0            = gridDim.x * blockDim.x;
+    size_t input_offset = static_cast<size_t>(gid1) * input_dim_size * static_cast<size_t>(stride);
```

Comment:
```suggestion
    size_t gid0            = blockIdx.x * blockDim.x + threadIdx.x;
    size_t gid1            = blockIdx.y * blockDim.y + threadIdx.y;
    size_t gsz0            = gridDim.x * blockDim.x;
    size_t input_offset = gid1 * input_dim_size * stride;
```
All those variables has always been casted to `size_t`

Reply:
change to size_t